clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -149,7 +149,24 @@ public FSDataInputStream open(Path path, int bufSize) throws IOException {
     @Override
     public RemoteIterator<LocatedFileStatus> listFiles(Path f, boolean recursive) throws FileNotFoundException, IOException {
         OPERATIONS_LOG.trace("list_files({}), recursive={}", f, recursive);
-        return toLocatedFileStatusIterator(new ListingIterator(f, recursive, listAmount));
+        return new RemoteIterator<LocatedFileStatus>() {
+            private final ListingIterator iterator = new ListingIterator(f, recursive, listAmount);
+
+            @Override
+            public boolean hasNext() throws IOException {
+                return iterator.hasNext();
+            }
+
+            @Override
+            public LocatedFileStatus next() throws IOException {
+                LakeFSFileStatus status = iterator.next();
+                BlockLocation[] locations = new BlockLocation[0];
+                if (status.isFile()) {
+                    locations = getFileBlockLocations(status, 0, status.getLen());
+                }
```

Comment:
This always allocates an empty array.  In Go that's almost fine, but it is a sin against the Java garbage collector.
```suggestion
                BlockLocation[] locations = status.isFile() ? getFileBlockLocations(status, 0, status.getLen()) : new BlockLocation[0];
```